main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"image"
+	"image/color"
+	"log"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	set "github.com/deckarep/golang-set"
+	"github.com/llgcode/draw2d/draw2dimg"
+)
+
+type LRTuple struct {
+	dir  float64
+	dist int
+}
+
+type Tuple struct {
+	rad float64
+	gx  int
+	gy  int
+}
+
+var ArrInput []LRTuple
+var xyInput []Tuple
+var iInput []Tuple
+var xyOutput []interface{}
+
+func printErr(err error) {
+	if err != nil {
+		log.Println(err.Error())
+	}
+}
+
+//parse and read textfile
+func readFile() {
+	file, err := os.Open("input.txt")
+	printErr(err)
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	result, err := reader.Read()
+	printErr(err)
+
+	log.Println(len(result))
+
+	for z := range result {
+		trimResult := strings.Trim(result[z], " ")
+		regStr := regexp.MustCompile("(\\D+)(\\d+)")
+		resultz := regStr.FindAllStringSubmatch(trimResult, -1)
+
+		headResult := resultz[0]
+
+		convDist, err := strconv.Atoi(headResult[2])
+		printErr(err)
+		radian := 0.00
+		if headResult[1] == "R" {
+			radian = (math.Pi / 180) * 90
+		} else {
+			radian = (math.Pi / 180) * -90
+		}
+
+		ArrInput = append(ArrInput, LRTuple{radian, convDist})
+
+	}
+	total := 0
+	for i := 0; i < len(ArrInput); i++ {
+		rads := strconv.FormatFloat(ArrInput[i].dir, 'E', -1, 64)
+		dist := strconv.Itoa(ArrInput[i].dist)
+
+		total = total + ArrInput[i].dist
+		log.Println("readFile: " + strconv.Itoa(i) + ", " + rads + ", " + dist)
+
+	}
+	log.Println(strconv.Itoa(total))
+
+}
+
+//calculate headings and steps
+func parseHeadings() {
+
+	first := Tuple{0.00, 0, 0}
+	xyInput = append(xyInput, first)
+
+	for bb := 0; bb < len(ArrInput); bb++ {
```

Comment:
I got no idea what you chose bb here. Similarly to before you can use range.

`for i, v := range ArrInput {`